common/tasks/letsencrypt.yaml

```diff
@@ -108,42 +129,209 @@
-    group: root
-    mode: 0700
+# remove
+# - name: install custom command to retrieve letsencrypt certificates
+#   template:
+#     src: usr_local_sbin_get_letsencrypt_cert.j2
+#     dest: /usr/local/sbin/get_letsencrypt_cert
+#     owner: root
+#     group: root
+#     mode: 0700
 
-- name: install the current SSL tar file
-  unarchive:
-    src: "{{ letsencrypt_archive }}/{{ i_service_connection.0.ipv4fqdn }}.tar.gz"
-    dest: /etc/
-  when: 
-  - initial_letsencrypt_status.stat.exists == False
-  - archive_status.stat.exists == True
+# part of the letsencrypt backup routine
+# - name: install the current SSL tar file
+#   unarchive:
+#     src: "{{ letsencrypt_archive }}/{{ i_service_connection.0.ipv4fqdn }}.tar.gz"
+#     dest: /etc/
+#   when:
+#   - initial_letsencrypt_status.stat.exists == False
+#   - archive_status.stat.exists == True
+# the behaviour of this tar file is not defined in the new certbot task
 
-- name: run /usr/local/sbin/get_letsencrypt_cert
-  shell: /usr/local/sbin/get_letsencrypt_cert
+- name: Set hostname to FQDN of machine
+  ansible.builtin.hostname:
+    name: "{{ i_service_connection.0.ipv4fqdn }}"
 
+# - name: run /usr/local/sbin/get_letsencrypt_cert
+#   shell: /usr/local/sbin/get_letsencrypt_cert
 
-- name: install cron task for letsencrypt
-  copy:
-    src: etc_cron.d_get_letsencrypt_cert
-    dest: /etc/cron.d/get_letsencrypt_cert
-    owner: root
-    group: root
-    mode: 0644
+# - name: install cron task for letsencrypt
+#   copy:
+#     src: etc_cron.d_get_letsencrypt_cert
+#     dest: /etc/cron.d/get_letsencrypt_cert
+#     owner: root
+#     group: root
+#     mode: 0644
+
+# certbot task
+
+# --- certbot install ---
+- name: Install snapd for certbot
+  ansible.builtin.dnf:
+    name: snapd
+    state: present
+
+- name: Enable snapd socket
+  ansible.builtin.systemd_service:
+    name: snapd.socket
+    enabled: true
+
+- name: Enable classic snap
+  ansible.builtin.file:
+    src: /var/lib/snapd/snap
+    dest: /snap
+    state: link
+
+- name: Ensure snapd socket started
+  ansible.builtin.systemd_service:
+    name: snapd.socket
+    state: started
+
+- name: Restart snapd service
+  ansible.builtin.systemd_service:
+    name: snapd.service
+    state: restarted
+
+- name: Restart snapd seeded service
+  ansible.builtin.systemd_service:
+    name: snapd.seeded.service
+    state: restarted
+
+- name: Install certbot
+  community.general.snap:
+    classic: true
+    name: certbot
+    state: present
+
+- name: Prepare certbot command
+  ansible.builtin.file:
+    src: /snap/bin/certbot
+    dest: /usr/bin/certbot
+    state: link
+# --- end certbot install ---
+
+- name: Register if each virtual_host certificate store exists
+  ansible.builtin.stat:
+    path: "/etc/letsencrypt/live/{{ item.virtualhost }}"
+  loop: "{{ i_virtualhosts }}"
+  register: i_virtualhosts_certificates_store
+
+- name: Populate service facts
+  ansible.builtin.service_facts:
+
+- name: Stop httpd service if present
```

Comment:
We should check if httpd is installed//running and if so we should run certbot with the apache flag instead of stopping the process. If it's not there then we can run as is now in this playbook.

Reply:
implemented in [e833f75](https://github.com/hep-gc/uvic-heprc-ansible-playbooks/pull/14/commits/e833f75e5c97c957fc1facfa2c1355862d4e37ac)